install/get-teamware.sh

```diff
@@ -0,0 +1,130 @@
+  CURL=("wget" "-O" "-")
+else
+  # Can't find curl, but we do have Docker so we can run curl via docker
+  CURL=("$DOCKER" "run" "--rm" "curlimages/curl" "-LO")
+fi
+
+# Gather information for env file
+set -a
+echo 'GATE Teamware needs to know the public host name at which it will be accessed'
+echo 'by users - only connections to that host name or "localhost" are accepted.'
+read -e -p 'Public hostname of GATE Teamware (e.g. teamware.example.com): ' DJANGO_ALLOWED_HOSTS
+read -e -p 'Email address of initial "admin" user [default "teamware@example.com"]: ' SUPERUSER_EMAIL
+read -e -p 'Initial password for "admin" user [default "password"]: ' SUPERUSER_PASSWORD
+
+echo ''
+echo 'GATE Teamware needs to be able to send outgoing email for user registration,'
+echo 'password resets, etc.  You can configure an SMTP server here, or edit your'
+echo 'settings later to use the GMail API.'
+read -e -p 'SMTP server hostname: ' DJANGO_EMAIL_HOST
+read -e -p 'SMTP server port number [default 587, may be 25 or 465]: ' DJANGO_EMAIL_PORT
+read -e -p 'SMTP server username, if required: ' DJANGO_EMAIL_HOST_USER
+read -e -s -p 'SMTP server password, if required: ' DJANGO_EMAIL_HOST_PASSWORD
+echo "" # read -s doesn't add a line break
+read -e -p 'Does SMTP server require a secure connection? [Y/n]: ' EMAIL_SECURE
+
+case $EMAIL_SECURE in
+  [Nn]*)
+    DJANGO_EMAIL_SECURITY=
+    ;;
+
+  *)
+    if [ "$DJANGO_EMAIL_PORT" = "465" ]; then
+      DJANGO_EMAIL_SECURITY=ssl
+    else
+      DJANGO_EMAIL_SECURITY=tls
+    fi
+    ;;
+esac
+
+set +a
+
+# Download install package
+if [ "$IMAGE_TAG" = "latest" -o "$IMAGE_TAG" = "dev" ]; then
+  "${CURL[@]}" https://github.com/GateNLP/gate-teamware/releases/latest/download/install.tar.gz | tar xzf -
+else
+  "${CURL[@]}" https://github.com/GateNLP/gate-teamware/releases/download/v${IMAGE_TAG}/install.tar.gz | tar xzf -
+fi
+
+# Create backup folder
+mkdir backups
+
+# Double check permissions on the files that will be bind-mounted
+chmod 755 nginx create-django-db.sh
+chmod 644 nginx/*.template
+
+echo "Generating configuration file"
+
+source ./generate-docker-env.sh
+
+cat <<EOF
+
+Your GATE Teamware installation is ready to use, your configuration has been
+written to the file ".env", which you can edit as required.
+
+To start up GATE Teamware, run:
+
+EOF
+if [ "${COMPOSE[0]/ /_}" = "${COMPOSE[0]}" ]; then
+  # docker or compose path does not contain any spaces
+  echo "${COMPOSE[*]} up -d"
+else
+  # path to docker contains spaces, so must be quoted
+  echo "\"${COMPOSE[0]}\" ${COMPOSE[*]:1} up -d"
+fi
+
+cat <<EOF
+
+By default the application is accessible at http://localhost:8076 - to change
+the port number you will need to edit docker-compose.yml.  We strongly
+recommend that public access be via secure HTTPS, for which you will need
+to deploy a separate front end proxy forwarding to the GATE Teamware
+application; configuration of such a proxy is beyond the scope of this quick
+start script.
```

Comment:
Should we include some text similar to this regarding proxy forwarding alongside the install instructions to forewarn users? Just so someone doesn't run this script and find this out right at the end?

Reply:
I've gone one better in my latest revision - I've added an optional add-on to the `docker-compose` stack that brings up an https reverse proxy on port 443.  I don't want to make that a standard part of the stack because it only works if the machine is open to the internet on 80/443 (because it uses ACME to obtain certificates) but it's a way to mark the path of least resistance.